pgjdbc/src/main/java/org/postgresql/core/TypeInfo.java

```diff
@@ -116,4 +116,23 @@ void addCoreType(String pgTypeName, Integer oid, Integer sqlType, String javaCla
    * @throws SQLException if something goes wrong
    */
   boolean requiresQuotingSqlType(int sqlType) throws SQLException;
+
+  /**
+   * To handle 32-bit integers as oids, we must communicate with PostgreSQL while they are non-negative.
```

Comment:
Can we change this comment to be more like the comment in pgdatabasemetadata. This one is difficult to read. I know what it means right now, but in 2 years it will be harder to grok. Same comment as the one below.

Reply:
thx

Reply:
I've updated the PR to address, and hopefully fix, these concerns.